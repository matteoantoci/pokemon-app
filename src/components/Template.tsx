import React from 'react'
import CssBaseline from '@material-ui/core/CssBaseline'
import Typography from '@material-ui/core/Typography'
import Container from '@material-ui/core/Container'
import styled from 'styled-components'
import { AppBar, IconButton } from '@material-ui/core'
import HomeIcon from '@material-ui/icons/Home'
import Toolbar from '@material-ui/core/Toolbar'
import { Flex } from './Flex'

const Copyright = () => (
  <Typography variant="body2" color="textSecondary" align="center">
    {`Copyright Â© Matteo Antoci ${new Date().getFullYear()}`}
  </Typography>
)

const Wrapper = styled(Flex)`
  flex-direction: column;
  min-height: 100vh;
`

const Main = styled(Container)`
  margin-top: 32px;
  margin-bottom: 32px;
`

const Footer = styled.footer`
  padding: 16px 0;
  margin-top: auto;
  background-color: white;
`

const HomeButton = styled(IconButton).attrs({
  edge: 'start',
  color: 'inherit',
})`
  margin-right: 8px !important;
`

type TemplateProps = {
  onGoBack: () => void
}

export const Template: React.FC<TemplateProps> = ({ children, onGoBack }) => (
  <Wrapper>
    <CssBaseline />
    <AppBar position="relative">
      <Toolbar>
        <HomeButton onClick={onGoBack} data-test="home-btn">
          <HomeIcon />
        </HomeButton>
        <Typography variant="h6" color="inherit" noWrap>
          PokemonIndex
        </Typography>
      </Toolbar>
    </AppBar>
    <Main component="main" maxWidth="md">
      {children}
    </Main>
    <Footer>
      <Container maxWidth="sm">
        <Copyright />
      </Container>
    </Footer>
  </Wrapper>
)
