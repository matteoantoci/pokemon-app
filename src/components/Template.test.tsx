import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Template } from './Template'

describe('Template', () => {
  const contents = 'contents'

  it('renders correctly', () => {
    const onGoBack = jest.fn()
    const component = renderer.create(<Template onGoBack={onGoBack}>contents</Template>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when clicking on home button', () => {
    it('triggers callback', () => {
      const onGoBack = jest.fn()
      const wrapper = shallow(<Template onGoBack={onGoBack}>contents</Template>)

      const ctaButton = wrapper.find('[data-test="home-btn"]')
      ctaButton.simulate('click')

      expect(onGoBack).toHaveBeenCalled()
    })
  })
})
