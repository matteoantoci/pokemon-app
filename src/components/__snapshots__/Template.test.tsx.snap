// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template renders correctly 1`] = `
<div
  className="sc-bwzfXH hJoDfx sc-bdVaJa kZAgjJ"
>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root sc-ifAKCX bEYIDc MuiIconButton-colorInherit MuiIconButton-edgeStart"
        data-test="home-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        PokemonIndex
      </h6>
    </div>
  </header>
  <main
    className="MuiContainer-root sc-htpNat ljHVDm MuiContainer-maxWidthMd"
  >
    contents
  </main>
  <footer
    className="sc-bxivhb cIAaJV"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright Â© Matteo Antoci 2019
      </p>
    </div>
  </footer>
</div>
`;
